crates/sui-core/src/consensus_adapter.rs

```diff
@@ -459,36 +474,65 @@ impl ConsensusAdapter {
     fn check_submission_wrt_connectivity_and_scores(
         &self,
         positions: Vec<AuthorityName>,
-    ) -> (usize, bool, AuthorityName) {
+    ) -> (usize, usize, String, AuthorityName) {
         let low_scoring_authorities = self.low_scoring_authorities.load().load_full();
         let first_submitter = *positions.first().unwrap_or(&AuthorityName::default());
         if low_scoring_authorities.get(&self.authority).is_some() {
-            return (positions.len(), true, first_submitter);
+            return (
+                positions.len(),
+                positions.len(),
+                "".to_string(),
+                first_submitter,
+            );
         }
         let initial_position = get_position_in_list(self.authority, positions.clone());
-
+        let mut disconnected = false;
```

Comment:
Alternatively you could call these two vars connected & high_scoring, and then just set them in the closure below and then just use the negation to set the reason at the end instead of having both disconnected/connected & high_scoring/low_scoring vars.